tests/test_models_layers.py

```diff
@@ -219,7 +219,7 @@ def forward(self) -> LayerRef:
   net_dict = net.make_root_net_dict()
   pprint(net_dict)
   assert net_dict["output"]["from"] == "sub/linear"
-  assert net_dict["sub"]["subnetwork"]["linear"]["from"] == "base:data:data"
+  assert net_dict["sub"]["subnetwork"]["linear"]["from"] == "data:data"
```

Comment:
But shouldn't it be `"base:data:data"`? It should get the extern data from the root net, not from the subnet.

Reply:
> Wait now that I see, it is also checked in another test. Does that mean we are changing something here from returnn? because there the `base:` prefix is not needed right?

It is needed. Why should it not be needed? You want to get `data` from the root net.

Reply:
You are absolutely right, I think I am too tired right now. I forgot about the possible `from:` which we do not use but which is a theoretical possibility replacing data:data. My bad.